types/prosemirror-view/index.d.ts

```diff
@@ -683,14 +687,22 @@ export interface NodeView<S extends Schema = any> {
   /**
    * Called when a DOM
    * [mutation](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)
-   * happens within the view. Return false if the editor should
+   * or a selection change happens within the view. When the change is
+   * a selection change, the record will have a `type` property of
+   * `"selection"` (which doesn't occur for native mutation records).
+   * Return false if the editor should re-read the selection or
    * re-parse the range around the mutation, true if it can safely be
    * ignored.
    */
-  ignoreMutation?: ((p: MutationRecord) => boolean) | null;
+  ignoreMutation?: ((p: MutationRecord & { readonly type: ProsemirrorMutationRecordType }) => boolean) | null;
```

Comment:
Unfortunately, this doesn't work as intended: the type `MutationRecord & { readonly type: ProsemirrorMutationRecordType }` is equivalent to `MutationRecord`, because all the values of this type must be objects whose `type` property has the intersection type `MutationRecord['type'] & ProsemirrorMutationRecordType`, which is equivalent to `MutationRecord['type']`. (Since in general, the type `A & (A | B)` is equivalent to `A` for any types `A` and `B`.)

A better type to use here would be `MutationRecord | { type: 'selection', target: Element }`. This is more accurate since the object constructed [in this line in ProseMirror's source](https://github.com/ProseMirror/prosemirror-view/blob/cbda274c7942000698a83b1aa8cc5db1a4c43266/src/domobserver.js#L118) has only `type` and `target` set, not other properties of `MutationRecord`.

Reply:
Okay that makes sense. I've updated it.